sotopia/experimental/agents/moderator.py

```diff
@@ -175,36 +190,45 @@ async def booting(self) -> None:
             self.current_agent_index += 1
 
     async def wrap_up_and_stop(self) -> None:
-        if self.push_to_db:
-            await self.save()
+        epilog = await self.save()
+        print("episode saved")
+        if self.will_eval:
+            epilog = await self.eval(epilog)
         await asyncio.sleep(0.5)
-        print("stopping all agents")
+        print("result of this episode:\n", epilog)
         await self.r.publish(
-            f"shutdown:{self.node_name}",
+            "shutdown:moderator",
             "shutdown",
         )
 
+    async def eval(self, epilog: EpisodeLog) -> EpisodeLog:
+        """
+        evaluate the episode
+        """
```

Comment:
@JXZhou0224 Perfectly understand there could be some problems of backaward compatiblity. I initiated some snippets of the codes for helping you finishing that. Let me know if you have any problems. There are some unique benefits of using the previous evaluators, thus I would like to push to that direction.